crypto/pkcs7/bio/md.c

```diff
@@ -0,0 +1,178 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+#include <errno.h>
+#include <openssl/buffer.h>
+#include <openssl/evp.h>
+#include <openssl/pkcs7.h>
+#include "../../internal.h"
+#include "../crypto/bio/internal.h"
+#include "../internal.h"
+
+// |md_write| and |md_read| both contribute to the digest, |md_gets| returns
+// digest contents
+static int md_write(BIO *h, char const *buf, int num);
+static int md_read(BIO *h, char *buf, int size);
+static int md_gets(BIO *h, char *str, int size);
+static long md_ctrl(BIO *h, int cmd, long arg1, void *arg2);
+static int md_new(BIO *h);
+static int md_free(BIO *data);
+
+
+static const BIO_METHOD methods_md = {
+    BIO_TYPE_MD, "message digest", md_write, md_read, /*puts*/ NULL,
+    md_gets,     md_ctrl,          md_new,   md_free, /*callback_ctrl*/ NULL,
+};
```

Comment:
NP: This would be easier to read if each field was on a separate line.

Reply:
I agree, but this was `clang-format`'s doing. I've been using it for formatting stuff in the `pkcs7/` module. If there's some set of flags/options that we prefer to use I'm happy to specify them.